bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -162,8 +162,12 @@ public void statusUpdated(final Thing thing, ThingStatusInfo thingStatus) {
                 Bridge bridge = (Bridge) thing;
                 for (Thing bridgeThing : bridge.getThings()) {
                     if (thingStatus.getStatus() == ThingStatus.ONLINE) {
-                        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(ThingStatus.ONLINE).build();
-                        setThingStatus(bridgeThing, statusInfo);
+                    	ThingStatusInfo bridgeThingStatus = bridgeThing.getStatusInfo(); 
+                    	if (bridgeThingStatus.getStatus() == ThingStatus.ONLINE || 
```

Comment:
perfect, then I remove it, as I interpreted that in a way that these
updates were wanted.

2015-09-10 13:16 GMT+02:00 Kai Kreuzer <notifications@github.com>:

> In
> bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java
> <https://github.com/eclipse/smarthome/pull/442#discussion_r39148506>:
>
> > @@ -162,8 +162,12 @@ public void statusUpdated(final Thing thing, ThingStatusInfo thingStatus) {
> >                  Bridge bridge = (Bridge) thing;
> >                  for (Thing bridgeThing : bridge.getThings()) {
> >                      if (thingStatus.getStatus() == ThingStatus.ONLINE) {
> > -                        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(ThingStatus.ONLINE).build();
> > -                        setThingStatus(bridgeThing, statusInfo);
> > +                    	ThingStatusInfo bridgeThingStatus = bridgeThing.getStatusInfo();
> > +                    	if (bridgeThingStatus.getStatus() == ThingStatus.ONLINE ||
>
> FTR, @marcelrv <https://github.com/marcelrv> is referring to my comment openhab/openhab2#357
> (comment)
> <https://github.com/openhab/openhab2/pull/357#discussion_r38522344>.
> But this was not about setting statuses for child devices of a bridge, but
> about setting the bridge status itself.
> In the case here, we are iterating over all child devices and I agree that
> it does not make sense to send an ONLINE status if they are already ONLINE,
> since we do not even know at this place whether this is true or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/smarthome/pull/442/files#r39148506>.
>